src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -51,7 +57,25 @@ protected ReindexJobRecord()
         }
 
         [JsonProperty(JobRecordProperties.MaximumConcurrency)]
-        public ushort MaximumConcurrency { get; private set; }
+        public ushort MaximumConcurrency
+        {
+            get
+            {
+                return maximumConcurrency;
+            }
+
+            set
+            {
+                if (value > 10)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(MaximumConcurrency));
```

Comment:
>ArgumentOutOfRangeException [](http://example.com/codeflow?start=30&length=27)

What would happened if we deserialize from database object which would throw this exception?


I would move validation inside constructor instead of having validation in each property setter.

Reply:
Moved the logic to constructor